src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -281,12 +273,24 @@ class MySimulation : public SimulationHelper<simDim>
     virtual void init()
     {
         namespace nvmem = PMacc::nvidia::memory;
+
+        DataConnector &dc = Environment<>::get().DataConnector();
+
         // create simulation data such as fields and particles
-        fieldB = new FieldB(*cellDescription);
-        fieldE = new FieldE(*cellDescription);
-        fieldJ = new FieldJ(*cellDescription);
+        auto fieldB = new FieldB( *cellDescription );
```

Comment:
only a note: `std::shared_ptr<FieldB> fieldB(new FieldB( *cellDescription ));` prevent that someone can use the plain pointer later in this method.